internal/pkg/service/stream/definition/repository/source/source_exists.go

```diff
@@ -0,0 +1,29 @@
+package source
+
+import (
+	serviceError "github.com/keboola/keboola-as-code/internal/pkg/service/common/errors"
+	"github.com/keboola/keboola-as-code/internal/pkg/service/common/etcdop/op"
+	"github.com/keboola/keboola-as-code/internal/pkg/service/stream/definition/key"
+)
+
+func (r *Repository) ExistsOrErr(k key.SourceKey) *op.TxnOp[op.NoResult] {
+	return op.Txn(r.client).
+		Merge(r.branches.ExistsOrErr(k.BranchKey)).
+		Merge(r.schema.Active().ByKey(k).Exists(r.client).
+			WithEmptyResultAsError(func() error {
+				return serviceError.NewResourceNotFoundError("source", k.SourceID.String(), "branch")
+			}),
+		).
+		FirstErrorOnly()
+}
+
+func (r *Repository) MustNotExists(k key.SourceKey) *op.TxnOp[op.NoResult] {
```

Comment:
Has to be exported? Used nly within the package.

Reply:
I will add test.